libraries/GCS_MAVLink/GCS_Common.cpp

```diff
@@ -5169,6 +5169,30 @@ MAV_RESULT GCS_MAVLINK::handle_do_set_safety_switch_state(const mavlink_command_
     }
 }
 
+MAV_RESULT GCS_MAVLINK::handle_do_set_sys_cmp_id(const mavlink_command_int_t &packet, const mavlink_message_t &msg)
+{
+    const uint8_t new_system_id = packet.param1;
+    const uint8_t new_component_id = packet.param2;
+    const uint8_t do_reboot = packet.param3;
+
+    if (new_system_id == 0) {
+        return MAV_RESULT_FAILED;
+    }
+
+    if (!AP_Param::set_and_save_by_name("SYSID_THISMAV", new_system_id)) {
+        return MAV_RESULT_FAILED;
+    }
+
+    if (new_component_id != 0 && !AP_Param::set_and_save_by_name("COMPID_THISMAV", new_component_id)) {
+        return MAV_RESULT_FAILED;
+    }
+
+    if (do_reboot) {
```

Comment:
We need toi reject this command if we are armed!

Reply:
Good catch! Moved to the top of the scope